playbooks/system-criticality.md

```diff
@@ -31,21 +31,24 @@ Retired systems, spikes, or time-bound experiments that aren't significantly use
+possible.
+
+|                   | Level 1                                                                                                                                                                                                    | Level 2                                                  | Level 3                                                                                                                                 |
+| ----------------- | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | -------------------------------------------------------- | --------------------------------------------------------------------------------------------------------------------------------------- |
+| Development       | READMEs contain up-to-date set-up instructions<br/>Code review on all [pull requests](/playbooks/engineer-workflow.md#pull-requests)<br/>New systems or significant new components undergo [tech review]() | &larr;ditto                                              | &larr;ditto +<br/>Production environment is replicable locally (TODO)                                                                   |
+| Testing           | Automated tests<br/>Tests are run on PRs and `master` by a Continuous Integration pipeline                                                                                                                 | &larr;ditto                                              | &larr;ditto +<br/>Test coverage tooling                                                                                                 |
+| Deployment        | Zero-downtime deployment                                                                                                                                                                                   | &larr;ditto +<br/>High-fidelity staging environment      | &larr;ditto +<br/>Deployment and orchestration by Kubernetes                                                                            |
+| Availability      | External availability monitoring                                                                                                                                                                           | &larr;ditto                                              | &larr;ditto                                                                                                                             |
+| Performance       |                                                                                                                                                                                                            |                                                          | Latency-based monitors and alerting (tailored to service)                                                                               |
```

Comment:
For level 3 services, is there a latency threshold that applies to every service? Should we have every service with p99 < 1s for instance ?

Reply:
I'd originally suggested <200ms average response time for APIs and <1s for interactive sites, but we couldn't agree about these in the discussion so left it that systems with latency alerting get to define the appropriate thresholds.

Force, for context, has p50 server-side latency of 1.25s and p99 of about 7s. The recent redesigns also shift a lot of work from the client to the server, so narrowly specifying server-side expectations might not capture important changes there.

However, I'd love for this to be more actionable somehow, such as maybe agreeing to standards for _new_ pages.